src/vs/code/electron-main/window.ts

```diff
@@ -566,14 +573,10 @@ export class CodeWindow implements ICodeWindow {
 			configuration['extensions-dir'] = cli['extensions-dir'];
 		}
 
-		if (cli) {
-			configuration['disable-extensions'] = cli['disable-extensions'];
-		}
-
 		configuration.isInitialStartup = false; // since this is a reload
 
 		// Load config
-		this.load(configuration, true);
+		this.load(configuration, true, cli);
```

Comment:
And here I would not pass over cli but just wether extensions are disabled or not.